osu.Game/Overlays/RankingsOverlay.cs

```diff
@@ -86,6 +98,11 @@ public RankingsOverlay()
         [BackgroundDependencyLoader]
         private void load()
         {
+            apiState.BindTo(api.State);
+            apiState.BindValueChanged(onlineStateChanged, true);
+
+            placeholderContainer.Child = errorPlaceholder;
```

Comment:
This can't work.

* In this particular case, `errorPlaceholder` is not even initialised yet, hence the error (`null Drawables may not be added to CompositeDrawable`).
* Even if it *were* initialised, the flow you're trying to do isn't correct. Drawables need to be loaded if they are to be shown (either by being in the drawable tree by the end of `load()`, or, if loading is required at a later point, via `LoadComponentAsync()`). Otherwise if you try to put them in the hierarchy, you'll get a possibly slightly-confusing error message (`Disposed drawables cannot be added to the scene graph`).

There is not much reason to ever create the login placeholder more than once. Therefore I would suggest to add it into the placeholder container already in the constructor, but start with it hidden by setting `Alpha = 0` on `placeholderContainer`.